### ISSUE TITLE
merge|join outputs in the wrong order

### ISSUE THREAD

**Issue by m-abs** [twigjs/twig.js#82]
From Twig's test/Twig/Tests/Fixtures/filters/merge.test

Template:
```
{{ items|merge({'bar': 'foo'})|join }}
{{ items|merge({'bar': 'foo'})|keys|join }}
{{ {'bar': 'foo'}|merge(items)|join }}
{{ {'bar': 'foo'}|merge(items)|keys|join }}
{{ numerics|merge([4, 5, 6])|join }}
```

Data:
```json
{ items: { foo: 'bar' }, numerics: [ 1, 2, 3 ] }
```

Expected result:
```
barfoo
foobar
foobar
barfoo
123456
```

Actual result:
```
foobar
barfoo
barfoo
foobar
123456
```

I think this is because the merge-filter uses "unshift" to add the keys to obj._keys and not push, because if I change it to push, the test is a success.

**Comment by plepe** [comment]
No, I would assume that the problem is in the parser. Check the output of:
```
INPUT: {{ { "foo": 1, "bar": 2, "baz": 3 }|json_encode }}
OUTPUT: {"baz":3,"bar":2,"foo":1}
```

I've noticed this when I wrote a custom function, which needs an object as input. The order of keys was inverted.

**Comment by dave-irvine** [comment]
I believe this was fixed in #279
